### ISSUE TITLE
Add callable kernel

### ISSUE THREAD

**Issue by Warvito** [Mind-the-Pineapple/sklearn-rvm#15]
Similar to https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html, it would be good to allow callable kernel to be more similar to SKLEARN

**Comment by PedroFerreiradaCosta** [comment]
I think it's already callable.

From https://github.com/scikit-learn/scikit-learn/blob/e5698bde9/sklearn/metrics/pairwise.py#L1820 :

"    metric : string, or callable
        The metric to use when calculating kernel between instances in a
        feature array. If metric is a string, it must be one of the metrics
        in pairwise.PAIRWISE_KERNEL_FUNCTIONS.
        If metric is "precomputed", X is assumed to be a kernel matrix.
        Alternatively, if metric is a callable function, it is called on each
        pair of instances (rows) and the resulting value recorded. The callable
        should take two rows from X as input and return the corresponding
        kernel value as a single number. This means that callables from
        :mod:`sklearn.metrics.pairwise` are not allowed, as they operate on
        matrices, not single samples. Use the string identifying the kernel
        instead."

**Comment by PedroFerreiradaCosta** [comment]
What do you think @Warvito ?
